### ISSUE TITLE
Transient error in extended tests

### ISSUE THREAD

**Issue by paulidale** [openssl/openssl#8209]
There is a transient error in one of the Travis extended tests.  Rerunning caused the test failure to disappear.  The relevant output is:
```
   # testing wap-wsg-idm-ecid-wtls1
=================================================================
==13741==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x60400009edb4 at pc 0x000000510464 bp 0x7fff7f7f2bd0 sp 0x7fff7f7f2bc8
WRITE of size 1 at 0x60400009edb4 thread T0
    #0 0x510463 in bn2binpad /home/travis/build/openssl/openssl/crypto/bn/bn_lib.c:453:15
    #1 0x510463 in BN_bn2bin /home/travis/build/openssl/openssl/crypto/bn/bn_lib.c:469
    #2 0x59a666 in bn_i2c /home/travis/build/openssl/openssl/crypto/asn1/x_bignum.c:110:9
    #3 0x501096 in asn1_ex_i2c /home/travis/build/openssl/openssl/crypto/asn1/tasn_enc.c:504:16
    #4 0x500d82 in asn1_i2d_ex_primitive /home/travis/build/openssl/openssl/crypto/asn1/tasn_enc.c:477:9
    #5 0x4fff14 in ASN1_item_ex_i2d /home/travis/build/openssl/openssl/crypto/asn1/tasn_enc.c
    #6 0x50062b in asn1_template_ex_i2d /home/travis/build/openssl/openssl/crypto/asn1/tasn_enc.c:344:12
    #7 0x50005a in ASN1_item_ex_i2d /home/travis/build/openssl/openssl/crypto/asn1/tasn_enc.c:182:13
    #8 0x4ff87d in asn1_item_flags_i2d /home/travis/build/openssl/openssl/crypto/asn1/tasn_enc.c:74:12
    #9 0x4e7a57 in test_builtin /home/travis/build/openssl/openssl/test/ecdsatest.c:347:19
    #10 0x4e9ca7 in run_tests /home/travis/build/openssl/openssl/test/testutil/driver.c:343:19
    #11 0x4ea281 in main /home/travis/build/openssl/openssl/test/testutil/main.c:30:15
    #12 0x2aee24553f44 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21f44)
    #13 0x41b09b in _start (/home/travis/build/openssl/openssl/test/ecdsatest+0x41b09b)

0x60400009edb4 is located 0 bytes to the right of 36-byte region [0x60400009ed90,0x60400009edb4)
allocated by thread T0 here:
    #0 0x4bc273 in malloc /local/mnt/workspace/tmp/ubuntu_rel/llvm/utils/release/final/llvm.src/projects/compiler-rt/lib/asan/asan_malloc_linux.cc:67:3
    #1 0x4e7545 in test_builtin /home/travis/build/openssl/openssl/test/ecdsatest.c:276:99
    #2 0x4e9ca7 in run_tests /home/travis/build/openssl/openssl/test/testutil/driver.c:343:19
    #3 0x4ea281 in main /home/travis/build/openssl/openssl/test/testutil/main.c:30:15
    #4 0x2aee24553f44 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21f44)

SUMMARY: AddressSanitizer: heap-buffer-overflow /home/travis/build/openssl/openssl/crypto/bn/bn_lib.c:453:15 in bn2binpad
Shadow bytes around the buggy address:
  0x0c088000bd60: fa fa fd fd fd fd fd fd fa fa 00 00 00 00 00 00
  0x0c088000bd70: fa fa 00 00 00 00 00 00 fa fa fd fd fd fd fd fd
  0x0c088000bd80: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
  0x0c088000bd90: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
  0x0c088000bda0: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
=>0x0c088000bdb0: fa fa 00 00 00 00[04]fa fa fa fd fd fd fd fd fd
  0x0c088000bdc0: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
  0x0c088000bdd0: fa fa fd fd fd fd fd fa fa fa fd fd fd fd fd fa
  0x0c088000bde0: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
  0x0c088000bdf0: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
  0x0c088000be00: fa fa fd fd fd fd fd fd fa fa fd fd fd fd fd fd
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==13741==ABORTING
../../util/shlib_wrap.sh ../../test/ecdsatest => 1
not ok 1 - running ecdsatest
```

The configdata dump is:
```
Command line (with current working directory = .):
    /usr/bin/perl ./Configure linux-x86_64 no-asm enable-asan enable-rc5 enable-md2 no-shared -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -D__NO_STRING_INLINES -Wno-unused-command-line-argument
Perl information:
    /usr/bin/perl
    5.18.2 for x86_64-linux-gnu-thread-multi
Enabled features:
    afalgeng
    aria
    asan
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    camellia
    capieng
    cast
    chacha
    cmac
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    gost
    hw(-.+)?
    idea
    makedepend
    md2
    md4
    mdc2
    multiblock
    nextprotoneg
    pinshared
    ocb
    ocsp
    pic
    poly1305
    posix-io
    psk
    rc2
    rc4
    rc5
    rdrand
    rfc3779
    rmd160
    scrypt
    seed
    siphash
    siv
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method
Disabled features:
    ktls                    [default] OPENSSL_NO_KTLS
    asm                     [option]  OPENSSL_NO_ASM
    crypto-mdebug           [default] OPENSSL_NO_CRYPTO_MDEBUG
    crypto-mdebug-backtrace [default] OPENSSL_NO_CRYPTO_MDEBUG_BACKTRACE
    devcryptoeng            [default] OPENSSL_NO_DEVCRYPTOENG
    dynamic-engine          [forced]  
    ec_nistp_64_gcc_128     [default] OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                     [default] OPENSSL_NO_EGD
    external-tests          [default] OPENSSL_NO_EXTERNAL_TESTS
    fuzz-libfuzzer          [default] OPENSSL_NO_FUZZ_LIBFUZZER
    fuzz-afl                [default] OPENSSL_NO_FUZZ_AFL
    heartbeats              [default] OPENSSL_NO_HEARTBEATS
    msan                    [default] OPENSSL_NO_MSAN
    sctp                    [default] OPENSSL_NO_SCTP
    shared                  [option]  
    ssl-trace               [default] OPENSSL_NO_SSL_TRACE
    ubsan                   [default] OPENSSL_NO_UBSAN
    unit-test               [default] OPENSSL_NO_UNIT_TEST
    weak-ssl-ciphers        [default] OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                    [default] 
    zlib-dynamic            [default] 
    ssl3                    [default] OPENSSL_NO_SSL3
    ssl3-method             [default] OPENSSL_NO_SSL3_METHOD
Config target attributes:
    AR => "ar",
    ARFLAGS => "r",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    CXX => "g++",
    CXXFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    aes_asm_src => "aes_core.c aes_cbc.c",
    aes_obj => "aes_core.o aes_cbc.o",
    apps_aux_src => "",
    apps_init_src => "",
    apps_obj => "",
    bf_asm_src => "bf_enc.c",
    bf_obj => "bf_enc.o",
    bn_asm_src => "bn_asm.c",
    bn_obj => "bn_asm.o",
    bn_ops => "SIXTY_FOUR_BIT_LONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cast_asm_src => "c_enc.c",
    cast_obj => "c_enc.o",
    cflags => "-pthread -m64",
    chacha_asm_src => "chacha_enc.c",
    chacha_obj => "chacha_enc.o",
    cmll_asm_src => "camellia.c cmll_misc.c cmll_cbc.c",
    cmll_obj => "camellia.o cmll_misc.o cmll_cbc.o",
    cppflags => "",
    cpuid_asm_src => "mem_clr.c",
    cpuid_obj => "mem_clr.o",
    cxxflags => "-std=c++11 -pthread -m64",
    defines => [  ],
    des_asm_src => "des_enc.c fcrypt_b.c",
    des_obj => "des_enc.o fcrypt_b.o",
    disable => [  ],
    dso_scheme => "dlfcn",
    ec_asm_src => "",
    ec_obj => "",
    enable => [ "afalgeng" ],
    ex_libs => "-ldl -pthread",
    includes => [  ],
    keccak1600_asm_src => "keccak1600.c",
    keccak1600_obj => "keccak1600.o",
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DOPENSSL_USE_NODELETE -DL_ENDIAN",
    lib_defines => [  ],
    md5_asm_src => "",
    md5_obj => "",
    modes_asm_src => "",
    modes_obj => "",
    module_cflags => "-fPIC",
    module_cxxflags => "",
    module_ldflags => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    multilib => "64",
    padlock_asm_src => "",
    padlock_obj => "",
    perl_platform => "Unix",
    perlasm_scheme => "elf",
    poly1305_asm_src => "",
    poly1305_obj => "",
    rc4_asm_src => "rc4_enc.c rc4_skey.c",
    rc4_obj => "rc4_enc.o rc4_skey.o",
    rc5_asm_src => "rc5_enc.c",
    rc5_obj => "rc5_enc.o",
    rmd160_asm_src => "",
    rmd160_obj => "",
    shared_cflag => "-fPIC",
    shared_defflag => "-Wl,--version-script=",
    shared_defines => [  ],
    shared_ldflag => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,-soname=",
    shared_target => "linux-shared",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",
    uplink_aux_src => "",
    uplink_obj => "",
    wp_asm_src => "wp_block.c",
    wp_obj => "wp_block.o",
Recorded environment:
    BUILDFILE = 
    CC = ccache clang
    CROSS_COMPILE = 
    CXX = 
    HASHBANGPERL = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = 
    RC = 
    WINDRES = 
    __CNF_CFLAGS = 
    __CNF_CPPDEFINES = 
    __CNF_CPPFLAGS = 
    __CNF_CPPINCLUDES = 
    __CNF_CXXFLAGS = 
    __CNF_LDFLAGS = 
    __CNF_LDLIBS = 
Makevars:
    AR              = ar
    ARFLAGS         = r
    CC              = ccache clang
    CFLAGS          = -Wall -O3 -Wno-unused-command-line-argument
    CPPDEFINES      = FUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION __NO_STRING_INLINES
    CPPFLAGS        = 
    CPPINCLUDES     = 
    CXX             = g++
    CXXFLAGS        = -Wall -O3 -Wno-unused-command-line-argument
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         = 
    LDLIBS          = 
    PERL            = /usr/bin/perl
    RANLIB          = ranlib
    RC              = windres
NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile
build file:
    Makefile
build file templates:
    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
    Configurations/common.tmpl
```

**Comment by mattcaswell** [comment]
Seems to have been happening ever since #7329. Only seems to affect curve wap-wsg-idm-ecid-wtls1

**Comment by mattcaswell** [comment]
The degree for this curve is 113 and therefore would have previously been filtered out by the check that was removed in #7329. I suspect that this curve does not in fact support ECDSA but I can't find out any information about it. A possible fix is to exclude it from the list of already excluded curves here:

https://github.com/openssl/openssl/blob/5a285addbf39f91d567f95f04b2b41764127950d/test/ecdsatest.c#L246-L247

It would be nice to confirm that this curve definitely doesn't support ECDSA though. Perhaps @bbbrumley can shed some light.

**Comment by bbbrumley** [comment]
```
$ openssl ecparam -name wap-wsg-idm-ecid-wtls1 -text -param_enc explicit -noout
Field Type: characteristic-two-field
Basis Type: tpBasis
Polynomial:
    02:00:00:00:00:00:00:00:00:00:00:00:00:02:01
A:    1 (0x1)
B:    1 (0x1)
Generator (uncompressed):
    04:01:66:79:79:a4:0b:a4:97:e5:d5:c2:70:78:06:
    17:00:f4:4b:4a:f1:ec:c2:63:0e:08:78:5c:eb:cc:
    15
Order: 
    00:ff:ff:ff:ff:ff:ff:ff:fd:bf:91:af:6d:ea:73
Cofactor:  2 (0x2)
```
Then with my group's tooling ...

```
sage: load_session('wap-wsg-idm-ecid-wtls1.sobj')
sage: hex(G[0].integer_representation())
'0x1667979a40ba497e5d5c270780617L'
sage: hex(G[1].integer_representation())
'0xf44b4af1ecc2630e08785cebcc15L'
sage: hex(order)
'fffffffffffffffdbf91af6dea73'
sage: hex(E.cardinality())
'1fffffffffffffffb7f235edbd4e6'
sage: hex(G.order())
'fffffffffffffffdbf91af6dea73'
sage: G.order().is_prime()
True
```

So ... technically should be OK for ECDSA from a functionality perspective. Gotta dig more and look at #7329.

**Comment by bbbrumley** [comment]
I'm trying to remember, it's been like 8 years. But is `ecdsatest.c` one of those tests that hijacks the RNG and feeds it known stuff to make ECDSA deterministic to pass KATs? If so, you cannot just change to `RAND_bytes(digest, SHA512_DIGEST_LENGTH))`.

**Comment by mattcaswell** [comment]
> But is `ecdsatest.c` one of those tests that hijacks the RNG and feeds it known stuff to make ECDSA deterministic to pass KATs?

It does - but not in this particular sub-test.

**Comment by bbbrumley** [comment]
It has been "transiently failing" for a while now. I managed to place an `assert` to catch it. Bisecting now. Seems unrelated to #7329.

**Comment by bbbrumley** [comment]
OK it's there since at least 1.1.0. Just a memory management bug in `ecdsatest.c`. I'll submit the PR now.

**Comment by levitte** [comment]
Did that resolve this issue, or is there more that keeps it open?

**Comment by mattcaswell** [comment]
#8232 got closed in favour of #8237 which is still waiting to be merged.
